zulip/zulip/__init__.py

```diff
@@ -1539,6 +1539,23 @@ def get_storage(self, request: Optional[Dict[str, Any]] = None) -> Dict[str, Any
             request=request,
         )
 
+    def remove_storage(self, **request: Any) -> Dict[str, Any]:
```

Comment:
I think we should type `request` properly, `Any` is not the way. Please go through [this](https://chat.zulip.org/#narrow/stream/92-learning/topic/mypy.20error/near/978210) message to understand how it should be done. Will highlight the same for the return type.

I think in this case it should be a `List[str]`, with a return of `Dict[str, str]`.

Moreover, I am not sure if we want to use `**request` as a param, `keys` I think would be better in terms of code clarity as we won't probably need more params for this function in the future.

```python
    return self.call_endpoint(
            url='bot_storage',
            method='DELETE',
            request={"keys": keys},
        )
```

Reply:
Hello @LoopThrough-i-j, sorry for the late response.
concerning the `**request` parameter and changing it to `keys`, do you mean something like this:
```
def remove_storage(self, keys: Optional[List[str]] = None) -> Dict[str, str]:
      
        return self.call_endpoint(
            url='bot_storage',
            method='DELETE',
            request={"keys": keys},
        )
```
And if so, why not just keep it at `request=keys` instead of `request={"keys": keys}`, to keep the consistency throughout the codebase, don't you think it will get a bit confusing to keep up on which function uses `keys=["entry 1"]` instead of `{'keys': ["entry 1"]}`  as a parameter, why make this the exception?

Reply:
I would like the keys to be:
```
keys: List[str]  # Not Optional
```
Why optional.

Moreover, I don't think it is reasonable to use `**request` on a function that can just have 1 parameter, just for consistency. Rather defining a proper parameter would be more reasonable.

> And if so, why not just keep it at request=keys instead of request={"keys": keys}
because that's how the `self.call_endpoint` works. It accepts a `Dict` and it is reasonable to do so.

Reply:
> don't you think it will get a bit confusing to keep up on which function uses keys=["entry 1"] instead of {'keys': ["entry 1"]} as a parameter

That's what the typing is for, and moreover, it won't be much of a difference from a user perspective.

Reply:
> why make this the exception?

This is not about making an exception, but about not following what is done everywhere just for consistency. Using the `keys` variable instead of `**request` will help the developers know exactly what are the arguments and what is getting passes in `request`, though this is not required or reasonable everywhere it makes sense here, knowing that there is not much scope for additional params.

Reply:
> Why optional.


well, if left empty it removes the entire bot storage, but if you don't want it optional how can we remove the entire bot storage?

Reply:
we can send an empty list.

Reply:
okay, or we can make this the parameter, what do you think?
`keys: List[str] = None`

Reply:
This will be a type error.

Reply:
```python
return self.call_endpoint(
            url='bot_storage',
            method='DELETE',
            request={"keys": keys} if keys != [] else None,
        )
```

Reply:
I tested out putting `keys: List[str] = []` and `keys: List[str] = None` and both of them work just fine, and i have noticed that other functions in the codebase uses None, without any issues, so i believe both of them work alike, what do you think?

Reply:
When Using `None` it is fine to use optional then, but I don't think we should make `keys: List[str] = None`, either use 
`keys: List[str] = []` or `keys: Optional[List[str]] = None`

Reply:
Still, I think sending an empty list is better, as the `bot_storage` API supports an empty list as well.

Reply:
Okay, I changed it to `keys: List[str] = []`, and changed the rest of the pr accordingly, I hope it meets your standards.

Reply:
This looks fine, thanks.

Reply:
These are not exactly my standards, just a small try to make sure the new code pushed in does not need reactors while cleaning up things in the future.